src/d3/hapticplot.d3.ts

```diff
@@ -108,12 +105,54 @@ export class Hapticplot{
    * @param hoverColor The color the point takes on while hovered
    * @param size The radius of the point being hovered
    */
-  private onHoverStart(point, hapticIntensity, hoverColor, size){
+  private onHoverStart(datum, point, hoverColor){
     d3.select(point)
       .attr('color', hoverColor);
     if (point.components?.sound?.isPlaying){
       point.components.sound.stopSound();
     }
+    d3.select(d3.event.detail?.hand)
+    .on('abuttonup',  (d, i, g) => this.speakPosition(datum, point))
+    .on('xbuttonup',  (d, i, g) => this.speakPosition(datum, point));
+  }
+
+  /**
+   * Reads out the position of the given point in spoken audio, and resets its sound component to
+   * the appropriate marimba note audio upon completion
+   * @param datum The datum from which the given points position and audio is generated
+   * @param point A data point entity
+   */
+  private speakPosition(datum, point){
+    const position = (d3.select(point).attr('position') as unknown) as Vector3;
+    const posString =
+      `X${Math.round((position.x - GRAPH_OFFSET.x) * 100) / 100}Y${Math.round((position.y - GRAPH_OFFSET.y) * 100) / 100}Z${Math.round((position.z - GRAPH_OFFSET.z) * 100) / 100}`;
```

Comment:
Any way to break this line up so it's not so long?

Reply:
yes! I split them into separate literals for each axis and then concatenate them.